src/block.rs

```diff
@@ -1,34 +1,118 @@
 //use std::stream::Stream;
 //use futures::io::AsyncRead;
 
-#[derive(Debug)]
+use serde::{Serialize, Deserialize};
+
+use crate::burnfee::BurnFee;
+
+#[derive(Serialize, Deserialize, PartialEq, Debug, Clone)]
 pub struct Block {
-    
+    pub body:     BlockBody,
+    pub is_valid: u8,
+    mintid:   u32,
+    maxtid:   u32,
+    bsh:      [u8; 32],
+}
+
+#[derive(Serialize, Deserialize, PartialEq, Debug, Clone)]
+pub struct BlockBody {
+    pub id:          u32,
+    pub ts:          u64,
+    pub prevbsh:     [u8; 32],
+    // pub creator:     PublicKey,
+    // pub txs:         Vec<Transaction>,
+    pub bf:	         BurnFee,
```

Comment:
Just as a flag, I'd highly recommend we keep the nomenclature for objects and variables the same between RS and JS and that it is changes from the existing nomenclature that need to be justified. Reason is so that development effort is easier to understand / switch between the two:

bf / ts / blk / etc.

The shorter terms have generally been chosen because we don't want big names for the JSON version (devs should be able to inspect the JSON versions and see what the variables are -- if we start having different names for RS / JS / JSON then things get very, very messy. So lets please flag and discuss changes as a group.